src/sumo_gym/envs/fmp.py

```diff
@@ -79,58 +79,115 @@ def __init__(
+                
+                # create new vertex with charging station's location
+                x_coord, y_coord = charging_station[1]
+                new_vtx = Vertex(x_coord, y_coord)
+                self.vertices.append(new_vtx)
+                
+                # keep a list of edge indices to delete later
+                edge_id = charging_station[2]
+                edge_idx_delete.append(self.edge_dict[edge_id])
+
+                # create two new edges
+                # first get the start and end vertices of the old edge
+                old_edge_start_idx = self.edges[self.edge_dict[edge_id]].start
+                old_edge_end_idx = self.edges[self.edge_dict[edge_id]].end
+
+                new_edge1 = Edge(old_edge_start_idx, vtx_counter)
+                new_edge2 = Edge(vtx_counter, old_edge_end_idx)
+                self.edges.append(new_edge1)
+                self.edges.append(new_edge2)
+
+                # instantiate new ChargingStation with location set to vtx_counter
+                self.charging_stations.append(ChargingStation(vtx_counter, 220, charging_station[3]))
+                
+                vtx_counter += 1
                 counter += 1
 
-                departures.append(self.edges[self.edge_dict[vehicle[1]]].start)
+            self.charging_stations = np.array(self.charging_stations)
+            self.vertices = np.array(self.vertices)
 
-            self.electric_vehicles = np.asarray(electric_vehicles)
-            self.departures = np.asarray(departures)
+            # remove indices from self.edges and fix self.edge_dict
+            # sort the indices in reverse order and pop the indices
+            # since the only elements in the list with different indices
+            # after a pop are those that reside on indices after the popped index
+            edge_idx_delete = sorted(list(set(edge_idx_delete)), reverse=True)
+            for idx in edge_idx_delete:
+                self.edges.pop(idx)
+            self.edges = np.array(self.edges)
 
-            demand = []
-            for d in raw_demand:
-                demand.append(Demand(self.vertex_dict[d[0]], self.vertex_dict[d[1]]))
+            # now, we need to repopulate self.edge_dict since the indices have changed
+            self.edge_dict = {} # sumo edge_id to idx in self.edges
+            counter = 0
+            for edge in self.edges:
+                if edge in self.revert_edge_dict:
+                    sumo_edge_id = self.revert_edge_dict[edge]
+                    self.edge_dict[sumo_edge_id] = counter
+                counter += 1
 
-            self.demand = np.asarray(demand)
+            self.electric_vehicles = []
+            self.ev_dict = {} # ev id to idx in self.electric_vehicles
+            counter = 0
+            for vehicle in raw_electric_vehicles:
+                self.electric_vehicles.append(ElectricVehicles(counter, vehicle[1], 220, vehicle[2]))
```

Comment:
quick question:
So we are setting default 220 indicator for both ev and charging_station, which means that we are able to route any ev to any cs in the model, will this affect the sumo simulation? like, although we route ev1 to cs2, it actually cannot be charged in cs2 in sumo, will this happen?

Reply:
I think Shuo said to ignore this for now but yeah setting the indicators to the same default value means any EV can go to any CS